### PR TITLE
[17.0] [MIG] sale_order_invoicing_grouping_criteria: Migration to 17.0

### PR DESCRIPTION
**Pull request by RogerSans**
supersede: https://github.com/OCA/account-invoicing/pull/1707
Standard migration
T-6282
@HaraldPanten

### CODE REVIEW

**Review by luis-ron — APPROVED** [review]
Functional Review: LGTM 👍🏻

File: sale_order_invoicing_grouping_criteria/README.rst [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+2. Locate inside "Sales Order Invoicing" section, the field "Default
+   Grouping Criteria".
+3. Introduce there the grouping criteria to be applied by default. If
+   empty, the general default of invoicing address + currency + company
+   will be applied.
+
+Usage
+=====
+
+1. Go to *Sales > Invoicing > Orders to Invoice*.
```
Comment (Tisho99) [inline sale_order_invoicing_grouping_criteria/README.rst:77]:
```suggestion
1. Go to *Sales > To Invoice > Orders to Invoice*.
```

File: sale_order_invoicing_grouping_criteria/README.rst [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+2. Select sales orders whose invoicing you want to do.
+3. Click on *Action > Invoice Order*.
```
Comment (Tisho99) [inline sale_order_invoicing_grouping_criteria/README.rst:79]:
```suggestion
3. Click on *Action > Create Invoices*.
```

**Review by Tisho99 — CHANGES_REQUESTED** [review]
Please, review the README, i think that the strings of the menus and the buttons  have changed in v17

Here are 2 examples

image

File: sale_order_invoicing_grouping_criteria/tests/test_sale_order_invoicing_group_criteria.py [hunk 3]
```diff
@@ -113,3 +114,8 @@ def test_commercial_field(self):
         self.assertEqual(
             children.sale_invoicing_grouping_criteria_id, self.grouping_criteria
         )
+
+    @classmethod
```
Comment (Tisho99) [inline sale_order_invoicing_grouping_criteria/tests/test_sale_order_invoicing_group_criteria.py]: Are you sure this changes are needed? I think that if the requests library was nos used before we don't have to add this code

Comment (Tisho99) [reply sale_order_invoicing_grouping_criteria/tests/test_sale_order_invoicing_group_criteria.py]: I refer to all changes done in tests

File: sale_order_invoicing_grouping_criteria/README.rst (hunk 1 above)
Comment (RogerSans) [reply sale_order_invoicing_grouping_criteria/README.rst:77]: Changed

File: sale_order_invoicing_grouping_criteria/README.rst (hunk 2 above)
Comment (RogerSans) [reply sale_order_invoicing_grouping_criteria/README.rst:79]: Changed

File: sale_order_invoicing_grouping_criteria/tests/test_sale_order_invoicing_group_criteria.py (hunk 3 above)
Comment (RogerSans) [reply sale_order_invoicing_grouping_criteria/tests/test_sale_order_invoicing_group_criteria.py]: removed

File: sale_order_invoicing_grouping_criteria/tests/test_sale_order_invoicing_group_criteria.py [hunk 4]
```diff
@@ -1,12 +1,13 @@
 # Copyright 2019 Tecnativa - Pedro M. Baeza
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo.tests import TransactionCase
+from odoo.tests import TransactionCase, requests
```
Comment (Tisho99) [inline sale_order_invoicing_grouping_criteria/tests/test_sale_order_invoicing_group_criteria.py]: You also need to remove this

File: sale_order_invoicing_grouping_criteria/tests/test_sale_order_invoicing_group_criteria.py [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
 
 
 class TestSaleOrderInvoicingGroupingCriteria(TransactionCase):
     @classmethod
     def setUpClass(cls):
+        cls._super_send = requests.Session.send
```
Comment (Tisho99) [inline sale_order_invoicing_grouping_criteria/tests/test_sale_order_invoicing_group_criteria.py]: You also need to remove this

**Review by Tisho99 — CHANGES_REQUESTED** [review]

File: sale_order_invoicing_grouping_criteria/tests/test_sale_order_invoicing_group_criteria.py (hunk 5 above)
Comment (RogerSans) [reply sale_order_invoicing_grouping_criteria/tests/test_sale_order_invoicing_group_criteria.py]: Done

File: sale_order_invoicing_grouping_criteria/tests/test_sale_order_invoicing_group_criteria.py (hunk 4 above)
Comment (RogerSans) [reply sale_order_invoicing_grouping_criteria/tests/test_sale_order_invoicing_group_criteria.py]: Done

**Review by Tisho99 — APPROVED** [review]
LGTM

**Review by HaraldPanten — COMMENTED** [review]
@RogerSans you forgot to add Sygel contributors. Didn't you?

**Comment by RogerSans** [comment]
> @RogerSans you forgot to add Sygel contributors. Didn't you?

Yes, my mistake. Added.

**Pull request merged**